packages/sdk/src/lib/dataProtectorSharing/addToCollection.ts

```diff
@@ -82,13 +82,15 @@ export const addToCollection = async ({
     protectedData: vProtectedData,
     sharingContractAddress,
   });
-  vOnStatusUpdate({
-    title: 'APPROVE_COLLECTION_CONTRACT',
-    isDone: true,
-    payload: {
-      approveTxHash: approveTx.hash,
-    },
-  });
+  if (!approveTx) {
+    vOnStatusUpdate({
+      title: 'APPROVE_COLLECTION_CONTRACT',
+      isDone: true,
+      payload: {
+        approveTxHash: approveTx?.hash,
+      },
+    });
+  }
```

Comment:
If I read the code well, there will be no 'APPROVE_COLLECTION_CONTRACT' status sent if there is an approve transaction. Why?

---

To make the condition a bit clearer, we can extract it as a variable:

```
const hasContractAlreadyBeenApproved = !approveTx
```

Or I'd suggest to add a comment:
```
  // If no transaction, it probably means the contract has already been approved
  if (!approveTx) {
    // ...
  }
```

Reply:
Fix commit : 57d976249a5ec253c2ebfa7300c24dea72568033